doc/connectivity/networking/api/mqtt_sn.rst

```diff
@@ -84,15 +84,31 @@ used. An example configuration for UDP transport is shown below:
 
    mqtt_sn_client_init(&client, &client_id, &tp.tp, evt_cb, tx_buf, sizeof(tx_buf), rx_buf, sizeof(rx_buf));
 
-After the configuration is set up, the MQTT-SN client can connect to the gateway.
-While the MQTT-SN protocol offers functionality to discover gateways through an
-advertisement mechanism, this is not implemented yet in the library.
-
-Call the ``mqtt_sn_connect`` function, which will send a ``CONNECT`` message.
-The application should periodically call the ``mqtt_sn_input`` function to process
-the response received. The application does not have to call ``mqtt_sn_input`` if it
-knows that no data has been received (e.g. when using Bluetooth). Note that
-``mqtt_sn_input`` is a non-blocking function, if the transport struct contains a
+After the configuration is set up, the network address for the gateway to
+connect to must be defined. The MQTT-SN protocol offers functionality to
+discover gateways through an advertisement or a search mechanism. A user
+should do at least one of the following steps to define a Gateway for the library:
+
+* Call the ``mqtt_sn_add_gw`` function to manually define a Gateway address.
+* Wait for a ``MQTT_SN_EVT_ADVERTISE``.
+* Call the ``mqtt_sn_search`` function and wait for a ``MQTT_SN_EVT_GWINFO`` callback.
+  Make sure to call the ``mqtt_sn_input`` function periodically to process incoming messages.
+
+Example ``mqtt_sn_search`` function call:
+
+.. code-block:: c
+
+	err = mqtt_sn_search(&mqtt_client, 1);
+	k_sleep(K_SECONDS(10));
+	err = mqtt_sn_input(&mqtt_client);
+	__ASSERT(err == 0, "mqtt_sn_search() failed %d", err);
+
+After the Gateway address has been defined or found, the MQTT-SN client can
+connect to the gateway. Call the ``mqtt_sn_connect`` function, which will send a
+``CONNECT`` message. The application should periodically call the ``mqtt_sn_input``
+function to process the response received. The application does not have to call
+``mqtt_sn_input`` if it knows that no data has been received (e.g. when using Bluetooth).
+Note that ``mqtt_sn_input`` is a non-blocking function, if the transport struct contains a
```

Comment:
```suggestion
* Call the :c:func:`mqtt_sn_add_gw` function to manually define a Gateway address.
* Wait for an :c:enumerator:`MQTT_SN_EVT_ADVERTISE`.
* Call the :c:func:`mqtt_sn_search` function and wait for an :c:enumerator:`MQTT_SN_EVT_GWINFO` callback.
  Make sure to call the :c:func:`mqtt_sn_input` function periodically to process incoming messages.

Example :c:func:`mqtt_sn_search` function call:

.. code-block:: c

	err = mqtt_sn_search(&mqtt_client, 1);
	k_sleep(K_SECONDS(10));
	err = mqtt_sn_input(&mqtt_client);
	__ASSERT(err == 0, "mqtt_sn_search() failed %d", err);

After the Gateway address has been defined or found, the MQTT-SN client can
connect to the gateway. Call the :c:func:`mqtt_sn_connect` function, which will send a
``CONNECT`` message. The application should periodically call the :c:func:`mqtt_sn_input`
function to process the response received. The application does not have to call
:c:func:`mqtt_sn_input` if it knows that no data has been received (e.g. when using Bluetooth).
Note that :c:func:`mqtt_sn_input` is a non-blocking function, if the transport struct contains a
```

BTW I appreciate that the rest of the file probably has other instances of C functions/enumerators incorreclty referenced, but at least it would be good to get these ones right (and maybe you'll feel like making a full pass on the file as a follow-up :-))

Reply:
Thanks for the feedback! I have never worked with RST files before so I am learning along the way. Made a full pass on the file and hope it is correct now! :)